golem-worker-service-base/tests/api_gateway_end_to_end_tests.rs

```diff
@@ -2490,6 +2512,15 @@ nUhg4edJVHjqxYyoQT+YSPLlHl6AkLZt9/n1NJ+bft0=
             all_signing_algs.clone(),
             Default::default(),
         )
+        .set_jwks({
+            let public_key = rsa::RsaPublicKey::from_public_key_pem(TEST_PUBLIC_KEY)
+                .expect("Failed to parse public key");
+
+            let n = public_key.n().to_bytes_be();
+            let e = public_key.e().to_bytes_be();
+            let kid = JsonWebKeyId::new("my-key-id".to_string());
+            JsonWebKeySet::new(vec![CoreJsonWebKey::new_rsa(n, e, Some(kid))])
+        })
```

Comment:
Now the test case is better, that the provider metadata will have the public key in it.
And the corresponding id_token will have its private key. Much easier to understand tests